### DISCUSSION TITLE
twinBasic Forms and UIAutomation

### DISCUSSION THREAD

**Discussion by mburns08109** [twinbasic/twinbasic#964] [General]
I have not explored this in much depth yet, but thanks to Mike Wolfe and, in turn, David Pineault's articles about compacting the currently open database (found here: https://www.devhut.net/access-vba-compact-the-current-database/), I've been introduced to the existence of the CUIAutomation class and its interface.
see: https://docs.microsoft.com/en-us/windows/win32/winauto/entry-uiautocore-overview

Now, my question is: will the twinBasic forms UI be accessible via the CUIAutomation class/interface (...since it's based on a browser-based forms engine)? ...or is there something that needs to be done to make them available via that API?

(To clarify, I'm not talking about the tbIDE - at least not _"per se"_, except in that its probably the best example of the sort of UI I'm thinking about being able to use/navigate via the UIAutomation API.)

**Comment by bclothier** [comment]
[Your tB application can allow automation if your application implements the required interfaces](https://docs.microsoft.com/en-us/windows/win32/winauto/uiauto-interfaces). It'd be up to Wayne whether he'd want to implement for tB IDE but it looks like extra work for little gain.

**Reply by mburns08109** [reply]
Actually Ben, I'm thinking that if the tB Forms engine generically does support UIAutomation, then that opens up tB apps to be supported by   automated testing suites later on. (Then, by default, so would the tBIDE - but that's really neither here nor there in terms of real importance.)
